ayeletK/assignment3/position.cpp

```diff
@@ -0,0 +1,19 @@
+class Position{
```

Comment:
Nice class/struct in order to be used inside a set.

Tip:
Since all its functions and fields are public, you can define this a `struct`.

Another tip:
An alternative to your own struct can be using the `std::pair` library class of C++.
Pair holds two values, in this case `std::pair<int, int>` would do the trick.

This has both advantages and disadvantages:

Reasons to use a pair:
1. Common to all readers - users know what a pair is even if they are new to your code
2. Compare and equality operators are built in (can be used in a set directly without additional code. Even in a hash set).

Reasons to use a dedicated struct (e.g. Position):
1. Using the struct name is more readable (The intent in `set<Position>` is clearer than `set<pair<int,int>>`).
2. You have control over the field names (you could call them row/col instead of first/second), more readable.

For completion, Java language has no pair-like class for this reason exactly: they prefer a dedicated class/struct for each particular case. In C++ we have the freedom to choose :)

Reply:
What about a combination of them both?
```c++
class Position{
  pair<int,int> _pos;
public:
  Position(int row, int col): pos(row,col) {}
  int get_row(){ return _pos.first; }
  int get_col(){ return _pos.second; }
  bool operator==(Position pos) { return this->_pos == pos._pos; }
  bool operator<(Position pos) { return this->_pos < pos._pos; }
};
```